src/NavierStokes.hpp

```diff
@@ -0,0 +1,355 @@
+#ifndef HH_NS_HH
+#define HH_NS_HH
+
+#include <deal.II/base/conditional_ostream.h>
+#include <deal.II/base/function.h>
+
+#include <deal.II/distributed/fully_distributed_tria.h>
+
+#include <deal.II/fe/fe_simplex_p.h>
+#include <deal.II/fe/fe_system.h>
+#include <deal.II/fe/fe_values.h>
+#include <deal.II/fe/fe_values_extractors.h>
+#include <deal.II/fe/mapping_fe.h>
+
+#include <deal.II/dofs/dof_handler.h>
+#include <deal.II/dofs/dof_renumbering.h>
+#include <deal.II/dofs/dof_tools.h>
+
+#include <deal.II/grid/grid_in.h>
+#include <deal.II/grid/grid_tools.h>
+
+#include <deal.II/lac/solver_cg.h>
+#include <deal.II/lac/solver_gmres.h>
+#include <deal.II/lac/trilinos_solver.h>
+#include <deal.II/lac/trilinos_block_sparse_matrix.h>
+#include <deal.II/lac/trilinos_parallel_block_vector.h>
+#include <deal.II/lac/trilinos_precondition.h>
+#include <deal.II/lac/trilinos_sparse_matrix.h>
+
+#include <deal.II/numerics/data_out.h>
+#include <deal.II/numerics/matrix_tools.h>
+#include <deal.II/numerics/vector_tools.h>
+
+#include <fstream>
+#include <iostream>
+#include <mpi.h>
+//#include <boost/range/adaptor/reversed.hpp>
+
+#include <chrono>
+
+
+using namespace dealii;
+
+
+template<size_t Dim>
+using FE_Type = FE_SimplexP<Dim>;
+
+template<size_t Dim>
+using Q_Type = QGaussSimplex<Dim>;
+
+class NavierStokes
+{
+public:
+  static constexpr unsigned int dim = DIM;
+
+  class ForcingTerm : public Function<dim>
+  {
+  public:
+    virtual double
+    value(const Point<dim> &/*p*/,
+      const unsigned int /*component*/ = 0) const override
+    {
+      return 0.0;
+    }
+  };
+
+  
+
+  class InletVelocity : public Function<dim>
+  {
+  public:
+    InletVelocity()
+      : Function<dim>(dim + 1)
+    {}
+
+
+    #ifndef NS_INPUT
```

Comment:
Using preprocessor macros for this conditional behavior should be your last resort, when all other options fail. Preprocessor macros may break C++ syntax and suffer from the same issues of global variables.

The alternatives include `if constexpr` statements, template specializations, or plain old `if` statements.